docs/source/plugin/bot.rst

```diff
@@ -335,3 +335,152 @@ You can also iterate over the list of users and filter them by privileges::
 
     Read about the :class:`~sopel.tools.target.Channel` and
     :class:`~sopel.tools.target.User` classes for more details.
+
+
+About time
+==========
+
+Your plugin may want to display dates and times in messages. For that, you can
+always count on the :mod:`datetime` built-in module. However, what if you would
+like to respect the date-format for a given user or a given channel? Various
+functions of :mod:`Sopel<sopel.tools.time>` can help you with that:
+
+* :func:`~sopel.tools.time.get_timezone` will fetch the right timezone for you
+* :func:`~sopel.tools.time.format_time` will format your aware datetime for you
+
+Here is a full example of that::
+
+    import datetime
+
+    import pytz
+
+    from sopel import plugin
+    from sopel.tools.time import format_time, get_timezone
+
+
+    @plugin.command('.t')
+    @plugin.require_chanmsg
+    def my_command(bot, trigger):
+        """Give time in a channel."""
+        time = pytz.UTC.localize(datetime.datetime.utcnow())
+        timezone = get_timezone(
+            bot.db,
+            bot.settings,
+            nick=trigger.nick,
+            channel=trigger.sender,
+        )
+        formatted_time = format_time(
+            bot.db,
+            bot.settings,
+            timezone,
+            trigger.nick,
+            trigger.sender,
+            time,
+        )
+        bot.say(formatted_time)
+
+Getting the time
+----------------
+
+As mentioned earlier, Sopel relies on a Python built-in module:
+:mod:`datetime`. This module allows you to get the current time like this::
+
+    >>> import datetime
+    >>> datetime.datetime.now()
+    datetime.datetime(2021, 7, 26, 18, 7, 13, 491786)
+    >>> datetime.datetime.utcnow()
+    datetime.datetime(2021, 7, 26, 16, 7, 16, 496404)
+
+As you can see at the moment of writing this documentation, there was a 2h
+offset between the local time and UTC. To properly manage timezones and UTC
+offsets, it is best to rely on ``pytz``, and work with UTC only::
+
+    >>> import pytz
+    >>> pytz.UTC.localize(datetime.datetime.utcnow())
+    datetime.datetime(2021, 7, 26, 16, 7, 19, 321828, tzinfo=<UTC>)
+
+This way, you'll always have an aware datetime to work with (store, compare,
+manipulate, etc.) that doesn't depend on your local time, and you'll only
+convert it to the proper timezone when you need to.
+
+.. note::
+
+    You should **always** work with aware datetime objects. It's also easier to
+    always work with UTC+0 datetime as input and in storage, then convert to
+    another timezone when displaying time to a user.
+
+Getting the timezone
+--------------------
+
+Sopel uses ``pytz`` to handle timezones to manipulate aware datetimes, i.e.
+datetimes with a timezone information such as the UTC Offset and DST
+(`Daylight Saving Time`__). When using this library, getting a timezone
```

Comment:
```suggestion
datetimes with timezone-related information such as the UTC Offset and DST
(`Daylight Saving Time`__) status. When using this library, getting a timezone
```

Not strictly necessary to go full "timezone-related", but the "a" is not valid syntax.

Might be smoother to link "DST" directly to the wiki page instead of stopping to define it in a parenthetical. (Could use [the `:abbr:` role](https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#role-abbr), though it would have to be tested; Furo doesn't appear to include that in [its Kitchen Sink](https://pradyunsg.me/furo/kitchen-sink/).)